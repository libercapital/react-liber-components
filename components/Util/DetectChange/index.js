"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _DetectChange["default"];
  }
});

var _DetectChange = _interopRequireDefault(require("./DetectChange"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }