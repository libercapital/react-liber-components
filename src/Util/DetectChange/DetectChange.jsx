
import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';

class DetectChange extends Component {
  shouldComponentUpdate(nextProps) {
    return nextProps.children !== this.props.children;
  }
  componentDidUpdate() {
    const { onChange } = this.props;
    onChange();
  }

  render() {
    return <Fragment>{this.props.children}</Fragment>;
  }
}

DetectChange.propTypes = {
  children: PropTypes.oneOfType([PropTypes.any]),
  onChange: PropTypes.func.isRequired,
};

DetectChange.defaultProps = {
  children: null,
};

export default DetectChange;
