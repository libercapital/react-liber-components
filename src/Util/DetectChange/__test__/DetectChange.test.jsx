import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import DetectChange from '../DetectChange';

describe('Testing NavBar Component', () => {
  let props;
  configure({ adapter: new Adapter() });
  beforeEach(() => {
    props = {
      onChange: sinon.spy(),
    };
  });
  it('should render correctly', () => {
    const wrapper = shallow(<DetectChange {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call function onChange when children change', () => {
    const wrapper = shallow(<DetectChange {...props} />);
    wrapper.setProps({ children: 'asdasd' });
    expect(props.onChange.called).toBeTruthy();
  });
});
