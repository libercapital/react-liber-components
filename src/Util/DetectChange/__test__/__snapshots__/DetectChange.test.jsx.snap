// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NavBar Component should render correctly 1`] = `<Fragment />`;
